zcash_client_sqlite/src/wallet/init/migrations/ephemeral_addresses.rs

```diff
@@ -0,0 +1,199 @@
+            wallet::TransparentAddressMetadata,
+        };
+        use zcash_keys::keys::UnifiedSpendingKey;
+        use zcash_primitives::{block::BlockHash, legacy::keys::NonHardenedChildIndex};
+        use zcash_protocol::consensus::Network;
+        use zip32::{fingerprint::SeedFingerprint, AccountId as Zip32AccountId};
+
+        use crate::{
+            error::SqliteClientError,
+            wallet::{
+                account_kind_code, init::init_wallet_db_internal, transparent::ephemeral, GAP_LIMIT,
+            },
+            WalletDb,
+        };
+
+        let network = Network::TestNetwork;
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), network).unwrap();
+
+        let seed0 = vec![0x00; 32];
+        init_wallet_db_internal(
+            &mut db_data,
+            Some(Secret::new(seed0.clone())),
+            &super::DEPENDENCIES,
+            false,
+        )
+        .unwrap();
+
+        let birthday = AccountBirthday::from_sapling_activation(&network, BlockHash([0; 32]));
+
+        // Simulate creating an account prior to this migration.
+        let account0_index = Zip32AccountId::ZERO;
+        let account0_seed_fp = [0u8; 32];
+        let account0_kind = account_kind_code(AccountSource::Derived {
+            seed_fingerprint: SeedFingerprint::from_seed(&account0_seed_fp).unwrap(),
+            account_index: account0_index,
+        });
+        assert_eq!(u32::from(account0_index), 0);
+        let account0_id = crate::AccountId(0);
+
+        let usk0 = UnifiedSpendingKey::from_seed(&network, &seed0, account0_index).unwrap();
+        let ufvk0 = usk0.to_unified_full_viewing_key();
+        let uivk0 = ufvk0.to_unified_incoming_viewing_key();
+
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (id, account_kind, hd_seed_fingerprint, hd_account_index, ufvk, uivk, birthday_height)
+                 VALUES (:id, :account_kind, :hd_seed_fingerprint, :hd_account_index, :ufvk, :uivk, :birthday_height)",
+                named_params![
+                    ":id": account0_id.0,
+                    ":account_kind": account0_kind,
+                    ":hd_seed_fingerprint": account0_seed_fp,
+                    ":hd_account_index": u32::from(account0_index),
+                    ":ufvk": ufvk0.encode(&network),
+                    ":uivk": uivk0.encode(&network),
+                    ":birthday_height": u32::from(birthday.height()),
+                ],
+            )
+            .unwrap();
+
+        // The `ephemeral_addresses` table is expected not to exist before migration.
+        assert_matches!(
+            ephemeral::first_unstored_index(&db_data.conn, account0_id),
+            Err(SqliteClientError::DbError(_))
+        );
+
+        let check = |db: &WalletDb<_, _>, account_id| {
+            eprintln!("checking {account_id:?}");
```

Comment:
Leaving in this `eprintln!` is intentional; otherwise, you would not be able to see which account had the failure just from the line number and message of the assertion. Output from a test is suppressed by default.